src/goto-symex/symex_goto.cpp

```diff
@@ -221,7 +221,14 @@ void goto_symext::symex_goto(statet &state)
   framet::goto_state_listt &goto_state_list =
     state.top().goto_state_map[new_state_pc];
 
-  goto_state_list.emplace_back(state);
+  if(new_guard.is_true())
```

Comment:
Add a comment noting in the must-take case we're free to move the state away, since it can't be used while the guard is false